### PR TITLE
EPAM-3: Aws task 3

### PR DESCRIPTION
**Pull request by musiichuk**
AWS Task 3

Link to FrontEnd's App PR - https://github.com/musiichuk/nodejs-aws-fe/pull/2 
Link to url of front end part - https://d15tir2c449mlg.cloudfront.net/

Get All Products - https://h5wzx2j3d5.execute-api.eu-west-1.amazonaws.com/dev/products/
Get Product By Id - https://h5wzx2j3d5.execute-api.eu-west-1.amazonaws.com/dev/products/7567ec4b-b10c-48c5-9345-fc73348a80a1
Not Found  - https://h5wzx2j3d5.execute-api.eu-west-1.amazonaws.com/dev/products/7567ec4b-b10c-48c5-9345-fc73348a80a113213
Not Found - https://h5wzx2j3d5.execute-api.eu-west-1.amazonaws.com/dev/products/123123

### CODE REVIEW

File: product-service/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "product-service",
+  "version": "1.0.0",
+  "description": "",
+  "main": "handler.js",
+  "dependencies": {
+    "babel": "^6.23.0",
+    "babel-core": "^6.26.3",
+    "babel-preset-es2015": "^6.24.1",
+    "jest": "^26.6.3"
```
Comment (artem-blazhko) [inline product-service/package.json:10]: Just FYI: those dependencies should be moved to "devDependencies"

**Review by artem-blazhko — APPROVED** [review]
Looks good to me, good work.

I do not see a link to swagger. Do you have it?

File: product-service/package.json (hunk 1 above)
Comment (musiichuk) [reply product-service/package.json:10]: Thanks, according to swagger, i don't have a url, but swagger doc was generated and it is available inside an app

File: product-service/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  },
+  "devDependencies": {
+    "@babel/preset-env": "^7.13.12",
+    "serverless-webpack": "^5.3.5",
+    "webpack": "^4.44.1",
+    "webpack-node-externals": "^2.5.0"
+  },
+  "scripts": {
```
Comment (wwrcth) [inline product-service/package.json:18]: Don't you want to have additional command to deploy? Will be much simpler :)

File: product-service/src/handlers/getProducts.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { getAllProducts } from '../services/getAllProducts/getAllProducts';
+import { buildHttpResponse } from '../utils/buildHttpResponse';
+
+export const getProducts = async () => {
```
Comment (wwrcth) [inline product-service/src/handlers/getProducts.js:4]: What if any error will be thrown? Will be a good idea to handle it with try catch block and return at least 500 error.

File: product-service/src/handlers/getProduct.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { buildHttpResponse } from '../utils/buildHttpResponse';
+
+export const getProduct = async (event, context) => {
+    const { id: productId = '' } = event.pathParameters;
+    try {
+      const product = await getProductById(productId);
+      if (!product) throw new Error('Product not found');
+      
+      return buildHttpResponse(200, product);
+    } catch (error) {
```
Comment (wwrcth) [inline product-service/src/handlers/getProduct.js:11]: What if we face any other error, except product not found. From my point of view, this case should be additionally protected with 500 error.

File: product-service/src/handlers/getProduct.test.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+const event = {
+    pathParameters: {
+        id: '7567ec4b-b10c-48c5-9345-fc73c48a8333',
+    }
+}
+
+describe('getProduct', () => {
+    test('should return statusCode 200', async () => {
+        const response = await getProduct(event);
+        expect(response.statusCode).toEqual(200);
```
Comment (NikitaHapeyeu) [inline product-service/src/handlers/getProduct.test.js:21]: any checking response body?

**Pull request merged**